### PR TITLE
[AIRFLOW-721] Descendant process can have disappeared before termination

### PR DESCRIPTION
**Pull request by bolkedebruin**
There is a race condition in helpers.py's kill_descendant_processes
that checks for running processes and then tries to terminate them.
This is not done atomically allowing for a small window where a PID
can disappear before termination.

Testing Done:
- Unittests should now pass again

@saguziel @aoen , please verify and see also my comment on https://github.com/apache/incubator-airflow/commit/23068924c09d6cebb0af6151b1950197ac16e67b

### CODE REVIEW

**Comment by bolkedebruin** [comment]
@alexvanboxel would you mind reviewing this plz? It should solve the issues we are seeing with Travis

**Comment by alexvanboxel** [comment]
Ah, that's why you get green PR. I'll schedule a review this evening.

File: airflow/utils/helpers.py [hunk 1]
```diff
@@ -205,10 +205,16 @@ def kill_descendant_processes(logger, pids_to_kill=None):
     logger.warn("Terminating descendant processes of {} PID: {}"
                 .format(this_process.cmdline(),
                         this_process.pid))
-    for descendant in descendant_processes:
+
+    temp_processes = descendant_processes
```
Comment (alexvanboxel) [inline airflow/utils/helpers.py]: Does this really take a copy?! Shouldn't it be something like this:
temp_processes = descendant_processes[:]

**Comment by alexvanboxel** [comment]
Aside from my remark it looks good

File: airflow/utils/helpers.py (hunk 1 above)
Comment (alexvanboxel) [reply airflow/utils/helpers.py]: @bolkedebruin just played around with some code an I think you really want to make a copy with:

`temp_processes = descendant_processes[:]`

If you fix it, I will merge the code in (I can now :-))

**Comment by bolkedebruin** [comment]
@alexvanboxel thanks! Should be ready now for merging so go ahead.

**Comment by plypaul** [comment]
Nice catch, and thanks for the fix @bolkedebruin !

**Pull request closed without merging**